extensions/amp-redbull-player/0.1/amp-redbull-player.js

```diff
@@ -0,0 +1,298 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  SandboxOptions,
+  createFrameFor,
+  isJsonOrObj,
+  objOrParseJson,
+  originMatches,
+} from '../../../src/iframe-video';
+import {Services} from '../../../src/services';
+import {VideoEvents} from '../../../src/video-interface';
+import {dict} from '../../../src/utils/object';
+import {disableScrollingOnIframe} from '../../../src/iframe-helper';
+import {getData, listen} from '../../../src/event-helper';
+import {installVideoManagerForDoc} from '../../../src/service/video-manager-impl';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {once} from '../../../src/utils/function';
+import {removeElement} from '../../../src/dom';
+import {userAssert} from '../../../src/log';
+import addAttributesToElement from './utils/addAttributesToElement';
+import getPlayerParams from './utils/getPlayerParams';
+
+/** @private @const */
+const TAG = 'amp-redbull-player';
+
+/** @private @const */
+const ANALYTICS_EVENT_TYPE_PREFIX = 'video-custom-';
+
+/** @private @const */
+const SANDBOX = [
+  SandboxOptions.ALLOW_SCRIPTS,
+  SandboxOptions.ALLOW_SAME_ORIGIN,
+  SandboxOptions.ALLOW_POPUPS,
+  SandboxOptions.ALLOW_POPUPS_TO_ESCAPE_SANDBOX,
+  SandboxOptions.ALLOW_TOP_NAVIGATION_BY_USER_ACTIVATION,
+];
+
+/**
+ * @return {!RegExp}
+ * @private
+ */
+const getAnalyticsEventTypePrefixRegex = once(
+  () => new RegExp(`^${ANALYTICS_EVENT_TYPE_PREFIX}`)
+);
+
+/** @implements {../../../src/video-interface.VideoInterface} */
+class AmpRedBullPlayer extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    const el = addAttributesToElement(element);
+    super(el);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {!UnlistenDef|null} */
+    this.unlistenFrame_ = null;
+
+    /** @private {string} */
+    this.tagId_ = null;
+
+    /** @private {string} */
+    this.locale_ = null;
+
+    /**
+     * @param {!Event} e
+     * @return {*} TODO(#23582): Specify return type
```

Comment:
Since this is new code then let's just specify it's return type. I think it's just `undefined`.

Reply:
done